### PR TITLE
Refresh the quickstart section

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Todos from @choldgraf list [here](https://github.com/jupyter/jupyter/issues/511#issuecomment-645075799):

- [ ]  Quickstart - page should provide more context for the different kinds of tools in the jupyter ecosystem, and note that "quickstart" means something different for each
    - [x]          Explain what "try.jupyter.org" is
    - [x] "Install" should be more like "install jupyter interfaces" instead of being notebook-specific.
    - [x] "Install" Fix the left TOC it's not showing up
    - [x] "Kernels" move the kernels section to a subsection of "Install"
    - [ ]  ~"Running" move this to a "getting started" section in the Jupyter Notebook docs, and link to that~
    - [x] "Migrating" move this somewhere else, maybe an "advanced" section in these docs? These docs probably aren't as necessary now but should be kept somewhere.

### CODE REVIEW

**Comment by choldgraf** [comment]
Nice! This is a great start! Let me know when you're ready for a review.

One other thought -> maybe we should rename this to "Get Started" rather than "Quickstart"? The former seems a little bit clearer to me. What do you think?

**Comment by GeorgianaElena** [comment]
> "Running" move this to a "getting started" section in the Jupyter Notebook docs, and link to that

I believe right now, the Jupyter Notebook docs links to the section here.
Since we renamed *quickstart* in *get started*, do you think it makes enough sense to leave the `running` section here?

**Comment by choldgraf** [comment]
yeah, let's leave it here for now and we can always move it later if it seems like the right choice.

File: docs/source/install.rst [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
+Please checkout the list and links below for additional information and instructions about
+how to get started with each of them.
+
+* **Jupyter Notebook** - *Web-based application for authoring documents that combine
+  live-code with narrative text, equations and visualizations.*
+
+  * `GitHub Repo <https://github.com/jupyter/notebook>`_
+  * `Docs <https://jupyter-notebook.readthedocs.io/en/latest/?badge=latest>`_
+
+* **JupyterLab** - *The next-generation web-based user interface for Project Jupyter.*
```
Comment (choldgraf) [inline docs/source/install.rst:21]:
```suggestion
* **JupyterLab** - *A next-generation web-based user interface for Project Jupyter.*
```

**Comment by choldgraf** [comment]
anything else to take care of here? Otherwise I'll merge this - we can always improve on the content later but this is a good step forward!

**Comment by GeorgianaElena** [comment]
I agree. Let's merge and continue improving through small PRs.

**Comment by choldgraf** [comment]
wohoo! thanks for this improvement @GeorgianaElena it is much appreciated :-) these docs could use some love and they are still read by a ton of people! (like, 40,000 a week) so this helps a lot

**Comment by GeorgianaElena** [comment]
> these docs could use some love

Then love shall them receive :heartbeat: 

> they are still read by a ton of people! (like, 40,000 a week)

Wow :heart:

**Pull request merged**